### PR TITLE
Simple GUI database exploring tool

### PR DESCRIPTION
**Pull request by MatthewHambley**
There is clearly much which could be done here but this gives an idea of what is possible. I'm afraid this branch is a bit of a beast. GUI coding always is. If we decide to go further with it we should first refactor it with a more rigorous Model/View/Control pattern.

### CODE REVIEW

File: TestCases/FortranSourceTree/working/program.f90 [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply TestCases/FortranSourceTree/working/program.f90:1]: No it isn't. Not sure how that got in there.

File: source/fab/entry.py [hunk 2]
```diff
@@ -123,5 +123,38 @@ def dump_entry() -> None:
...
+    return parser.parse_args()
+
+
+def fab_explorer() -> None:
+    """
+    Entry point for database exploration tool.
+    """
+    arguments = explorer_cli()
+
+    # TODO: We probably need a better default here.
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply source/fab/entry.py:151]: Indeed, but I'd like to hold making that change for a separate branch.

File: source/fab/explorer.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+from fab.language.fortran import FortranWorkingState
+
+
+class ExplorerWindow(tk.Frame):
+    """
+    Main window of the database explorer.
+    """
+    def __init__(self, state: StateDatabase):
+        self._root = tk.Tk()
+        self._root.title("Fab Database Explorer")
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply source/fab/explorer.py:25]: It should now expand properly.

File: source/fab/explorer.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+        self._unit_name = UnitNameFrame(self, fortran_db)
+        self._unit_name.grid(row=0, column=0)
+
+        self._unit_filename = UnitFileFrame(self, fortran_db)
+        self._unit_filename.grid(row=0, column=1)
+
+        self._unit_details = UnitInfoFrame(self, fortran_db)
+        self._unit_details.grid(row=0, column=2)
+
+        message = "Single-click to select, double-click to jump"
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply source/fab/explorer.py:147]: I had already implemented a pointer change to indicate which columns could be double-clicked but neglected to mention it.

**Comment by MatthewHambley** [comment]
This branch is ready for further review.

File: source/fab/explorer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (stevewardle) [reply source/fab/explorer.py:25]: And indeed it does!  The whole thing is massively improved, nice work

**Pull request merged**